singularity-eos/eos/get_sg_eos_functors.hpp

```diff
@@ -71,31 +73,89 @@ struct init_functor {
     for (int m = 0; m < nmat; ++m) {
       frac_mass_v(i, m) /= mass_sum;
     }
-    /* set inputs */
+    check_all_vals(i);
+    // count the number of participating materials and zero the inputs
     npte = 0;
     for (int m = 0; m < nmat; ++m) {
       if (frac_mass_v(i, m) > 1.e-12) {
+        // participating materials are those with non-negligible mass fractions
         pte_idxs(tid, npte) = eos_offsets_v(m) - 1;
         pte_mats(tid, npte) = m;
         npte += 1;
       } else {
         frac_ie_v(i, m) = 0.0;
       }
+      // zero the inputs
       vfrac_pte(tid, m) = 0.0;
       sie_pte(tid, m) = 0.0;
       temp_pte(tid, m) = 0.0;
       press_pte(tid, m) = 0.0;
     }
+    // Populate the inputs with consistent values.
+    // NOTE: the volume fractions and densities need to be consistent with the
+    // total specific volume since they are used to calculate internal
+    // quantities for the PTE solver
     for (int mp = 0; mp < npte; ++mp) {
       const int m = pte_mats(tid, mp);
-      rho_pte(tid, mp) = npte / spvol_v(i) * frac_mass_v(i, m);
-      vfrac_pte(tid, mp) = 1.0 / npte;
+      // Need to guess volume fractions
+      vfrac_pte(tid, mp) = frac_mass_v(i, m);
```

Comment:
FYI - I'm changing the initial guess for volume fractions from equipartitioning of volume irrespective of mass fractions. This guess is equivalent to assuming that all material densities are the same.

This got me further in my test problem than assuming that all volume fractions are the same. In the future I'll see if I can work on getting the volume fractions from the previous cycle in here and then requiring that they be well-defined.

Reply:
Interesting... good to know!